master/internal/configpolicy/postgres_task_config_policy.go

```diff
@@ -88,6 +90,43 @@ func GetTaskConfigPolicies(ctx context.Context,
 	return &tcp, nil
 }
 
+// GetPriorityLimit reads the priority limit for the given scope and workload type.
+// It returns found=false if no limit exists.
+func GetPriorityLimit(ctx context.Context, scope *int, workloadType string) (limit int, found bool, err error) {
+	if !ValidWorkloadType(workloadType) {
+		return 0, false, fmt.Errorf("invalid workload type: %s", workloadType)
+	}
+
+	wkspQuery := wkspIDQuery
+	if scope == nil {
+		wkspQuery = wkspIDGlobalQuery
+	}
+
+	var constraints model.Constraints
+	var constraintsStr string
+	err = db.Bun().NewSelect().
+		Table("task_config_policies").
+		Column("constraints").
+		Where(wkspQuery, scope).
+		Where("workload_type = ?", workloadType).
+		Scan(ctx, &constraintsStr)
+
+	if err == sql.ErrNoRows {
+		return 0, false, nil
+	} else if err != nil {
+		return 0, false, fmt.Errorf("error retrieving priority limit: %w", err)
+	}
+
+	if err = json.Unmarshal([]byte(constraintsStr), &constraints); err != nil {
+		return 0, false, err
+	}
```

Comment:
Are we sure that this won't error out when the string input into the db is a YAML and not a JSON?
Can we add a test for this? Our test cases seem to input the config policies in JSON format

Reply:
With how the PUT API is implemented, I think it's fair to assume that data in the database is in json-format. I'm glad you pointed this out - it was a good thing to check! As it turns out, the PUT call will error if we attempt to write yaml-formatted data to the task config policies table.

Reply:
Awesome! Thanks for looking into this 😆